tower-http/src/services/fs/serve_dir.rs

```diff
@@ -113,7 +125,14 @@ impl<ReqBody> Service<Request<ReqBody>> for ServeDir {
                 });
 
             let file = File::open(full_path).await?;
-            Ok(Output::File(file, mime))
+            let mut headers: HeaderMap<HeaderValue> = HeaderMap::new();
+            headers.insert(header::CONTENT_TYPE, mime);
+            headers.insert(
+                header::CACHE_CONTROL,
+                HeaderValue::from_str(&format!("max-age={}", max_age)).unwrap(),
+            );
```

Comment:
I think it would be nicer to just have the two headers as separate fields on `Output::File`. Then we don't need to allocate a `HeaderMap` and can add the headers one by one in the future.